src/conn/conn_tiered.c

```diff
@@ -0,0 +1,328 @@
+/*-
+ * Copyright (c) 2014-2020 MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "wt_internal.h"
+
+#ifdef __GNUC__
+#if __GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ > 1)
+/*
+ * !!!
+ * GCC with -Wformat-nonliteral complains about calls to strftime in this file.
+ * There's nothing wrong, this makes the warning go away.
+ */
+#pragma GCC diagnostic ignored "-Wformat-nonliteral"
+#endif
+#endif
+
+/*
+ * __tier_storage_once --
+ *     Perform one iteration of tiered storage maintenance.
+ */
+static void
+__tier_storage_once(WT_SESSION_IMPL *session, bool force)
+{
+    WT_UNUSED(session);
+    WT_UNUSED(force);
+    /*
+     * - See if there is any "merging" work to do to prepare and create an object that is
+     *   suitable for placing onto tiered storage.
+     * - Do the work to create said objects.
+     * - Move the objects.
+     */
+    return;
+}
+
+/*
+ * __tier_storage_remove_tree --
+ *     Perform one iteration of tiered storage local tier removal.
+ */
+static int
+__tier_storage_remove_tree(WT_SESSION_IMPL *session, const char *uri, bool force)
```

Comment:
Hmm, at the moment I think of this as removing the entire local table, not individual data items. This is an internal-to-this-file function, I would rather describe what it does in the name than something more generic. If we change what we do, wouldn't it be clearer to just change the function name? I'm not sure why we care about keeping the name the same for an internal-only function.

Please push back if you feel strongly because I don't feel strongly. `tier_storage_remove_local` is fine by me if you prefer it.

Reply:
I missed the detail that this function is intended for internal use only (even though that's what the example code does).  I was thinking that all the tiered_storage functions are public APIs.  This is fine.

Reply:
@sueloverso, I also think the naming is confusing.  I think `tier_storage_remove_local` is a better choice.